Gems/Atom/Tools/ShaderManagementConsole/Scripts/GenerateShaderVariantListUtil.py

```diff
@@ -16,6 +16,114 @@
+                    if has_repeat:
+                        continue
+
+                    shaderVarintIdDict[shaderAssetId].append(shaderVariantId)
+                    shaderOptionGroupsDict[shaderAssetId].append(shaderItem.GetShaderOptionGroup())
+
+        progressDialog.setValue(i)
+        if progressDialog.wasCanceled():
+            return
+
+    progressDialog.close()
+
+    # Generate the shader variant list for each shader asset
+    for shaderAssetId in shaderOptionGroupsDict:
+
+        shaderAssetInfo = asset.AssetCatalogRequestBus(
+            azlmbr.bus.Broadcast, 
+            'GetAssetInfoById', 
+            shaderAssetId   
+        )
+        fullShaderPath = azlmbr.shadermanagementconsole.ShaderManagementConsoleRequestBus(
+            azlmbr.bus.Broadcast, 
+            'GetFullSourcePathFromRelativeProductPath', 
+            shaderAssetInfo.relativePath
+        )
+        relativeShaderPath = azlmbr.shadermanagementconsole.ShaderManagementConsoleRequestBus(
+            azlmbr.bus.Broadcast, 
+            'GenerateRelativeSourcePath', 
+            fullShaderPath
+        )
+        shaderVariantList = azlmbr.shader.ShaderVariantListSourceData()
+        shaderVariantList.shaderFilePath = relativeShaderPath
+        shaderVariants = []
+        stableId = 1
+        for shaderOptionGroup in shaderOptionGroupsDict[shaderAssetId]:
+            variantInfo = azlmbr.shader.ShaderVariantInfo()
+            variantInfo.stableId = stableId
+            options = {}
+
+            shaderOptionDescriptors = shaderOptionGroup.GetShaderOptionDescriptors()
+            for shaderOptionDescriptor in shaderOptionDescriptors:
+                optionName = shaderOptionDescriptor.GetName()
+                optionValue = shaderOptionGroup.GetValueByOptionName(optionName)
+                if not optionValue.IsValid():
+                    continue
+
+                valueName = shaderOptionDescriptor.GetValueName(optionValue)
+                options[optionName] = valueName
+
+            if len(options) != 0:
+                variantInfo.options = options
+                shaderVariants.append(variantInfo)
+                stableId += 1
+
+        if shaderVariants:
+            shaderVariantList.shaderVariants = shaderVariants
+            pre, ext = os.path.splitext(relativeShaderPath)
+            if "Intermediate Assets" in fullShaderPath:
+                projectShaderVariantListFilePath = os.path.join(azlmbr.paths.projectroot, PROJECT_SHADER_VARIANTS_FOLDER, "Cache", "Intermediate Assets", f'{pre}.shadervariantlist')
+            else:
+                projectShaderVariantListFilePath = os.path.join(azlmbr.paths.projectroot, PROJECT_SHADER_VARIANTS_FOLDER, f'{pre}.shadervariantlist')
+            
+            # clean previously generated shader variant list file so they don't clash.
+            if os.path.exists(projectShaderVariantListFilePath):
+                os.remove(projectShaderVariantListFilePath)
```

Comment:
This could be done in a separate PR, but it would be nice to show the user a dialog with a table of info about all the .shadervariantlist files that are about to be saved, which ones are replacing existing files, and they can uncheck any .shadervariantlist to skip it.